sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/MainActivity.java

```diff
@@ -92,41 +71,42 @@ protected void onCreate(Bundle savedInstanceState) {
                 String json = "{ \"number\": 10 }";
                 Attachment attachment = new Attachment(json.getBytes(), "log.json");
                 scope.addAttachment(attachment);
-                scope.addAttachment(new Attachment(file.getPath()));
               });
+
+          Sentry.captureMessage("Some message with attachments.");
         });
 
     binding.captureException.setOnClickListener(
-        view ->
-            Sentry.captureException(
-                new Exception(new Exception(new Exception("Some exception.")))));
+        view -> Sentry.captureException(new Exception(new Exception("Some exception."))));
 
     binding.breadcrumb.setOnClickListener(
         view -> {
           Sentry.addBreadcrumb("Breadcrumb");
-          Sentry.setExtra("extra", "extra");
-          Sentry.setFingerprint(Collections.singletonList("fingerprint"));
-          Sentry.setTransaction("transaction");
-          Sentry.captureException(new Exception("Some exception with scope."));
-        });
-
-    binding.unsetUser.setOnClickListener(
-        view -> {
-          Sentry.setTag("user_set", "null");
-          Sentry.setUser(null);
+          Sentry.captureMessage("Some message with breadcrumbs.");
         });
 
     binding.setUser.setOnClickListener(
         view -> {
-          Sentry.setTag("user_set", "instance");
           User user = new User();
           user.setUsername("username_from_java");
-          // works with some null properties?
-          // user.setId("id_from_java");
           user.setEmail("email_from_java");
-          // Use the client's IP address
+          // Use the client's IP address or enable 'sendDefaultPii' that does it automatically.
           user.setIpAddress("{{auto}}");
           Sentry.setUser(user);
+          // pass null to Sentry#setUser to unset the user
+          Sentry.captureMessage("Some message with user.");
+        });
+
+    binding.localScope.setOnClickListener(
+        view -> {
+          Sentry.withScope(
```

Comment:
Since Hub is global, access to it from another thread would race to the data changed in this scope so I would rather avoid adding this example. We should consider adding the overloads to capture like we have on Cocoa and Flutter and bring up to the SDK evolution that we want to drop this methods.